### PR TITLE
Add REST implementation for Validator's `GetBeaconBlock`

### PR DESCRIPTION
**Pull request by PatriceVignola**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

It adds the `GetBeaconBlock` REST API endpoint for the validator.

**Other notes for review**

I noticed that the beacon node returns `PHASE0`, `ALTAIR` and `BELLATRIX` as the versions instead of `phase0`, `altair` and `bellatrix`. I believe this is because the ToString that protobuf generates matches the enums. The [spec](https://ethereum.github.io/beacon-APIs/#/Validator/produceBlockV2) says that the lowercase version should be returned, but it doesn't say if it's meant to be case sensitive. Either way, in this PR I handle both situations (lowercase and uppercase) to be compatible with both versions, but if it is a bug, it should be addressed in a different PR since it could touch a lot of files.

Also, the response `Eth-Consensus-Version` header doesn't seem to be sent by the beacon node, which is a requirement of the spec to facilitate unmarshaling.

### CODE REVIEW

**Comment by rkapka** [comment]
Both enum casing and the `Eth-Consensus-Version` header are handled through the API Middleware, [here](https://github.com/prysmaticlabs/prysm/blob/62ffed7861b2b97ac8eba48308c996469e1baa91/api/gateway/apimiddleware/process_field.go#L167-L170) and [here](https://github.com/prysmaticlabs/prysm/blob/9c21809c5061a7938fcc69b1b1ae06cd6b3abfbc/beacon-chain/rpc/apimiddleware/custom_handlers.go#L315) respectively.

**Comment by PatriceVignola** [comment]
> Both enum casing and the `Eth-Consensus-Version` header are handled through the API Middleware, [here](https://github.com/prysmaticlabs/prysm/blob/62ffed7861b2b97ac8eba48308c996469e1baa91/api/gateway/apimiddleware/process_field.go#L167-L170) and [here](https://github.com/prysmaticlabs/prysm/blob/9c21809c5061a7938fcc69b1b1ae06cd6b3abfbc/beacon-chain/rpc/apimiddleware/custom_handlers.go#L315) respectively.

The header is only for SSZ though no? And the struct doesn't seem to have the `enum` tag for the version, so I don't think it gets applied. I originally tried getting the header and the non-lowercase version, which I how I realized it wasn't working as I was expecting.

**Comment by rkapka** [comment]
The header makes sense only for SSZ, in JSON you have the `version` field. But you are correct that it's missing the `enum` tag. Do you mind fixing it here?

File: validator/client/beacon-api/beacon_block_proto_helpers.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+		}
+
+		attestationData, err := convertAttestationDataToProto(jsonAttestation.Data)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get attestation data")
+		}
+
+		signature, err := hexutil.Decode(jsonAttestation.Signature)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get attestation signature `%s`", jsonAttestation.Signature)
```
Comment (rkapka) [inline validator/client/beacon-api/beacon_block_proto_helpers.go:192]:
```suggestion
			return nil, errors.Wrapf(err, "failed to decode attestation signature `%s`", jsonAttestation.Signature)
```
Just for uniformity.

**Review by rkapka — APPROVED** [review]

**Pull request merged**